test/e2e/cluster_pullsecret.go

```diff
@@ -0,0 +1,323 @@
+			Expect(err).NotTo(HaveOccurred())
+
+			By("deploying the demo cluster bicep template")
+			_, err = framework.CreateBicepTemplateAndWait(ctx,
+				tc.GetARMResourcesClientFactoryOrDie(ctx).NewDeploymentsClient(),
+				*resourceGroup.Name,
+				"pull-secret-cluster",
+				framework.Must(TestArtifactsFS.ReadFile("test-artifacts/generated-test-artifacts/demo.json")),
+				map[string]interface{}{
+					"persistTagValue": false,
+					"clusterName":     customerClusterName,
+				},
+				45*time.Minute,
+			)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("getting credentials")
+			adminRESTConfig, err := framework.GetAdminRESTConfigForHCPCluster(
+				ctx,
+				tc.Get20240610ClientFactoryOrDie(ctx).NewHcpOpenShiftClustersClient(),
+				*resourceGroup.Name,
+				customerClusterName,
+				10*time.Minute,
+			)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("ensuring the cluster is viable")
+			err = verifiers.VerifyHCPCluster(ctx, adminRESTConfig)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("creating kubernetes client")
+			kubeClient, err := kubernetes.NewForConfig(adminRESTConfig)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("creating test pull secret")
+			username := "test-user"
+			auth := base64.StdEncoding.EncodeToString([]byte(username + ":" + testPullSecretPassword))
+
+			testPullSecret, err := framework.CreateTestDockerConfigSecret(
+				testPullSecretHost,
+				username,
+				testPullSecretPassword,
+				testPullSecretEmail,
+				pullSecretName,
+				pullSecretNamespace,
+			)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("creating the test pull secret in the cluster")
+			_, err = kubeClient.CoreV1().Secrets(pullSecretNamespace).Create(ctx, testPullSecret, metav1.CreateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			By("waiting for HCCO to merge the additional pull secret with the global pull secret")
+			Eventually(func() error {
+				return verifiers.VerifyPullSecretMergedIntoGlobal(testPullSecretHost).Verify(ctx, adminRESTConfig)
+			}, 300*time.Second, 15*time.Second).Should(Succeed(), "additional pull secret should be merged into global-pull-secret by HCCO")
+
+			By("verifying the DaemonSet for global pull secret synchronization is created")
+			Eventually(func() error {
+				return verifiers.VerifyGlobalPullSecretSyncer().Verify(ctx, adminRESTConfig)
+			}, 60*time.Second, 10*time.Second).Should(Succeed(), "global-pull-secret-syncer DaemonSet should be created")
+
+			By("verifying the pull secret was merged into the global pull secret")
+			err = verifiers.VerifyPullSecretAuthData(
+				"global-pull-secret",
+				pullSecretNamespace,
+				testPullSecretHost,
+				auth,
+				testPullSecretEmail,
+			).Verify(ctx, adminRESTConfig)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("reading pull-secret file from aro-hcp-qe-pull-secret directory")
+			pullSecretFilePath := filepath.Join("/var/run/aro-hcp-qe-pull-secret", "pull-secret")
+			pullSecretFileData, err := os.ReadFile(pullSecretFilePath)
```

Comment:
should we make this a framework function?  

If someone runs the test suite locally, this will fail.

Reply:
Moving this into a library function would make more sense if we could reconfigure the location of the aro-hcp-qe-pull-secret for local runs.

Reply:
This is in the framework now and we can specify an env var. for local runs